### PR TITLE
Update com.networknt:json-schema-validator dependency from 1.0.81 to 1.0.83

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
#7809 failed so updates dependency `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83 manually

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add  `com.github.luben:zstd-jni` version 1.5.5-3 ([#2996](https://github.com/opensearch-project/OpenSearch/pull/2996))
 - OpenJDK Update (April 2023 Patch releases) ([#7344](https://github.com/opensearch-project/OpenSearch/pull/7344)
 - Bump `com.amazonaws` 1.12.270 to `software.amazon.awssdk` 2.20.55 ([7372](https://github.com/opensearch-project/OpenSearch/pull/7372/))
+- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83
```
Comment (scrawfor99) [inline CHANGELOG.md:49]:
```suggestion
- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83 ([7833](https://github.com/opensearch-project/OpenSearch/pull/7933/))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add  `com.github.luben:zstd-jni` version 1.5.5-3 ([#2996](https://github.com/opensearch-project/OpenSearch/pull/2996))
 - OpenJDK Update (April 2023 Patch releases) ([#7344](https://github.com/opensearch-project/OpenSearch/pull/7344)
 - Bump `com.amazonaws` 1.12.270 to `software.amazon.awssdk` 2.20.55 ([7372](https://github.com/opensearch-project/OpenSearch/pull/7372/))
+- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83 ([7833](https://github.com/opensearch-project/OpenSearch/pull/7933/))
```
Comment (scrawfor99) [inline CHANGELOG.md:49]:
```suggestion
- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83 ([7933](https://github.com/opensearch-project/OpenSearch/pull/7933/))
```

**Comment by reta** [comment]
@scrawfor99 this one is problematic - there is a change in behaviour and our rest-api specs are failing validation

**Comment by scrawfor99** [comment]
Hi @reta, I noticed that too. Do you still want me to go forward with the update or did you prefer to wait? I can work on fixing it if you need.

**Comment by reta** [comment]
> Do you still want me to go forward with the update or did you prefer to wait? I can work on fixing it if you need.

I quickly checked what the cause / change was, I was not able to quickly identify the reason and abandoned (for the time being), if you have time to look, was be awesome

**Comment by scrawfor99** [comment]
Have gone through all the schema settings and believe the issue is because of changes here: https://github.com/networknt/json-schema-validator/pull/783/files#diff-776968543c85221a4907857a99fb117ab32872c5f0c020a717c11fe8162ef310. 

Going to look into how to fix this.

**Comment by reta** [comment]
> Going to look into how to fix this.

I suspected that as well, but what distracted me is that

```
this.hasEndAnchor = '$' == regex.charAt(regex.length() - 1);
```

where the `$` is checked at the end of the regex but the errors we have

```
[validate JSON][ERROR][rank_eval.json][$.rank_eval.params.expand_wildcards.options[0]: does not match the regex pattern ^\d*|[a-zA-Z_]+$]
```

do have `$` at the beginning (but I have not finished the investigation)

**Comment by scrawfor99** [comment]
HI @reta,

I ended up asking the maintainers of the dependency: https://github.com/networknt/json-schema-validator/issues/814#issuecomment-1581325261.

They said: 

"
I wrote some unit-tests and was able to reproduce the issue with options but not the issue with slices. The problem with options is that the pattern is not matching the end anchor when using the JDK's pattern. This does not occur when using Joni's regex. You can switch to Joni by setting SchemaValidatorsConfig.setEcma262Validator(true). I'll continue investigating why the end anchor does not match.
"

For now, I am going try to resolve it as they suggest or looking into converting.

**Comment by reta** [comment]
> For now, I am going try to resolve it as they suggest or looking into converting.

Thanks a lot @scrawfor99 !

File: buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java [hunk 3]
```diff
@@ -113,7 +114,7 @@ public void validate(InputChanges inputChanges) throws IOException {
                 File file = fileChange.getFile();
                 if (file.isDirectory() == false) {
                     // validate all files and hold on to errors for a complete report if there are failures
-                    getLogger().debug("Validating JSON [{}]", file.getName());
+                    getLogger().debug("Validating JSON: " + file.getName());
```
Comment (reta) [inline buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java:117]: Curious why this change? The previous version was more "lazy" and is not doing anything if the debug logging is not enabled.

Comment (scrawfor99) [reply buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java:117]: Oops, it is different because I had swapped to COUT for for reading the info and missed swapping it. I will fix that now.

Comment (scrawfor99) [inline buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java:117]:
```suggestion
                   getLogger().debug("Validating JSON [{}]", file.getName());
```

File: buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java [hunk 4]
```diff
@@ -113,7 +114,7 @@ public void validate(InputChanges inputChanges) throws IOException {
                 File file = fileChange.getFile();
                 if (file.isDirectory() == false) {
                     // validate all files and hold on to errors for a complete report if there are failures
-                    getLogger().debug("Validating JSON [{}]", file.getName());
+                   getLogger().debug("Validating JSON [{}]", file.getName());
```
Comment (scrawfor99) [inline buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java:117]:
```suggestion
                    getLogger().debug("Validating JSON [{}]", file.getName());
```

File: buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java [hunk 5]
```diff
@@ -143,7 +144,7 @@ public void validate(InputChanges inputChanges) throws IOException {
         }
     }
 
-    private void maybeLogAndCollectError(Set<ValidationMessage> messages, Map<File, Set<String>> errors, File file) {
+    private void maybeLogAndCollectError(Set<ValidationMessage> messages, Map<File, Set<String>> errors, File file) throws IOException {
```
Comment (scrawfor99) [inline buildSrc/src/main/java/org/opensearch/gradle/precommit/ValidateJsonAgainstSchemaTask.java:147]:
```suggestion
    private void maybeLogAndCollectError(Set<ValidationMessage> messages, Map<File, Set<String>> errors, File file) {
```

File: rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json [hunk 6]
```diff
@@ -36,11 +36,7 @@
       },
       "refresh":{
         "type":"enum",
-        "options":[
-          "true",
-          "false",
-          "wait_for"
-        ],
+        "options": ["true", "false", "wait_for"],
```
Comment (scrawfor99) [inline rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json:39]:
```suggestion
         "options":[
          "true",
          "false",
          "wait_for"
        ],
```

File: rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json [hunk 7]
```diff
@@ -36,7 +36,7 @@
       },
       "refresh":{
         "type":"enum",
-        "options":[
+         "options":[
```
Comment (scrawfor99) [inline rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json:39]:
```suggestion
        "options":[
```

**Comment by scrawfor99** [comment]
Apologies for the inline changes...

**Comment by reta** [comment]
Needs rebase :(

File: .ci/bwcVersions [hunk 8]
```diff
@@ -19,4 +19,5 @@ BWC_VERSION:
   - "2.7.0"
   - "2.7.1"
   - "2.8.0"
+  - "2.8.1"
```
Comment (reta) [inline .ci/bwcVersions]: Probably shouldn't be in this pull request (already in `main`)

File: CHANGELOG.md [hunk 9]
```diff
@@ -91,7 +91,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Security
 
 ## [Unreleased 2.x]
```
Comment (reta) [inline CHANGELOG.md]: Needs some cleanup ...

Comment (scrawfor99) [reply CHANGELOG.md]: Github is doing some weird stuff with the rebases. I think my branch is behind my main and then when I rebased it got confused. Sorry for all the notifications I am sure it is spamming...

File: CHANGELOG.md [hunk 10]
```diff
@@ -46,8 +46,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add  `com.github.luben:zstd-jni` version 1.5.5-3 ([#2996](https://github.com/opensearch-project/OpenSearch/pull/2996))
 - OpenJDK Update (April 2023 Patch releases) ([#7344](https://github.com/opensearch-project/OpenSearch/pull/7344)
 - Bump `com.amazonaws` 1.12.270 to `software.amazon.awssdk` 2.20.55 ([7372](https://github.com/opensearch-project/OpenSearch/pull/7372/))
+- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83 ([7933](https://github.com/opensearch-project/OpenSearch/pull/7933/))
 - Bump `com.google.http-client:google-http-client:1.43.2` from 1.42.0 to 1.43.2 ([7928](https://github.com/opensearch-project/OpenSearch/pull/7928)))
 - Bump `com.azure:azure-storage-blob` from 12.21.1 to 12.22.2 ([7930](https://github.com/opensearch-project/OpenSearch/pull/7930))
+- Bump `com.networknt:json-schema-validator` from 1.0.81 to 1.0.83
```
Comment (reta) [inline CHANGELOG.md:52]: Is it a duplicate entry?

Comment (scrawfor99) [reply CHANGELOG.md:52]: Oops, I didn't realize I already added it and must have missed it. Good catch.

**Review by reta — APPROVED** [review]
